### ISSUE TITLE
Cloning cluster doesn't complete

### ISSUE THREAD

**Issue by hackdna** [galaxyproject/cloudman#42]
The process appears to hang with the following messages in the CloudMan log:
```
2016-03-01 22:40:53,458 DEBUG           misc:205  Checking if bucket 'cm-3e863b9434af2d0e55c834cf200efc50' exists... it does not.
2016-03-01 22:40:53,458 DEBUG           misc:217  Creating bucket 'cm-3e863b9434af2d0e55c834cf200efc50'.
2016-03-01 22:40:53,612 DEBUG           misc:219  Created bucket 'cm-3e863b9434af2d0e55c834cf200efc50'.
2016-03-01 22:40:53,661 DEBUG           misc:591  Retrieved file 'shared/2016-03-01--22-05/shared_instance_file_list.txt' from bucket 'cm-5b88aa0dbe0f7fc5765ccdc7dc116101' on host 's3.amazonaws.com' to 'shared_instance_file_list.txt'.
2016-03-01 22:40:53,671 DEBUG           misc:643  Establishing handle with key object 'shared/2016-03-01--22-05/persistent_data.yaml'
2016-03-01 22:40:53,671 DEBUG           misc:647  Copying file 'cm-5b88aa0dbe0f7fc5765ccdc7dc116101/shared/2016-03-01--22-05/persistent_data.yaml' to file 'cm-3e863b9434af2d0e55c834cf200efc50/persistent_data.yaml'
2016-03-01 22:40:53,876 DEBUG           misc:643  Establishing handle with key object 'shared/2016-03-01--22-05/cm.tar.gz'
2016-03-01 22:40:53,876 DEBUG           misc:647  Copying file 'cm-5b88aa0dbe0f7fc5765ccdc7dc116101/shared/2016-03-01--22-05/cm.tar.gz' to file 'cm-3e863b9434af2d0e55c834cf200efc50/cm.tar.gz'
2016-03-01 22:40:54,259 DEBUG           misc:643  Establishing handle with key object 'shared/2016-03-01--22-05/cm_boot.py'
2016-03-01 22:40:54,259 DEBUG           misc:647  Copying file 'cm-5b88aa0dbe0f7fc5765ccdc7dc116101/shared/2016-03-01--22-05/cm_boot.py' to file 'cm-3e863b9434af2d0e55c834cf200efc50/cm_boot.py'
2016-03-01 22:40:54,454 DEBUG           misc:591  Retrieved file 'persistent_data.yaml' from bucket 'cm-3e863b9434af2d0e55c834cf200efc50' on host 's3.amazonaws.com' to 'shared_p_d.yaml'.
2016-03-01 22:40:54,462 DEBUG         master:1618 Initializing Galaxy cluster type from shared cluster
2016-03-01 22:40:54,693 INFO          master:1644 Created a data volume 'vol-153123ca' of size 200GB from shared cluster's snapshot 'snap-497a6b49'
2016-03-01 22:40:54,693 DEBUG         master:1650 Dumping scpd to file cm_cluster_config.yaml (which will become persistent_data.yaml): {'placement': 'us-east-1b', 'persistent_data_version': 3, 'cluster_type': u'Galaxy', 'deployment_version': 2, 'filesystems': [{'kind': 'snapshot', 'mount_point': '/mnt/galaxyIndices', 'ids': ['snap-7b992575'], 'roles': ['galaxyIndices'], 'name': 'galaxyIndices'}, {'kind': 'volume', 'mount_point': '/mnt/galaxy', 'ids': [u'vol-153123ca'], 'roles': ['galaxyTools', 'galaxyData'], 'name': 'galaxy'}], 'cluster_name': u'cloudman-test', 'machine_image_id': 'ami-d5246abf', 'services': [{'name': 'Postgres', 'roles': ['Postgres']}, {'name': 'ProFTPd', 'roles': ['ProFTPd']}, {'name': 'GalaxyReports', 'roles': ['GalaxyReports']}, {'name': 'Slurmd', 'roles': ['Slurmd']}, {'name': 'Nginx', 'roles': ['Nginx']}, {'name': 'NodeJSProxy', 'roles': ['NodeJSProxy']}, {'name': 'Supervisor', 'roles': ['Supervisor']}, {'name': 'Slurmctld', 'roles': ['Slurmctld', 'Job manager']}, {'home': '/mnt/galaxy/galaxy-app', 'name': 'Galaxy', 'roles': ['Galaxy']}], 'cluster_storage_type': u'volume'}
2016-03-01 22:40:54,729 DEBUG           misc:615  Saved file 'cm_cluster_config.yaml' of size 929B as 'persistent_data.yaml' to bucket 'cm-3e863b9434af2d0e55c834cf200efc50'
2016-03-01 22:40:54,776 DEBUG           misc:591  Retrieved file 'persistent_data.yaml' from bucket 'cm-3e863b9434af2d0e55c834cf200efc50' on host 's3.amazonaws.com' to 'pd.yaml'.
2016-03-01 22:40:54,883 DEBUG         master:483  Checking for and adding any previously defined cluster services
2016-03-01 22:40:54,884 DEBUG         master:489  Processing filesystems in an existing cluster config
2016-03-01 22:40:54,884 DEBUG         master:960  Trying to discover any volumes attached to this instance...
2016-03-01 22:40:55,275 DEBUG         master:977  Attached volumes: [Volume:vol-e80c1e37]
2016-03-01 22:40:55,276 DEBUG            ec2:389  Adding tag 'clusterName:cloudman-dev' to resource 'vol-e80c1e37'
2016-03-01 22:40:55,356 DEBUG     filesystem:32   Instantiating Filesystem object galaxyIndices with service roles: 'galaxyIndices'
2016-03-01 22:40:55,356 DEBUG         master:584  Checking if vol 'vol-e80c1e37' is file system 'galaxyIndices'
2016-03-01 22:40:55,356 DEBUG            ec2:414  Getting tag 'clusterName' on resource 'vol-e80c1e37'
2016-03-01 22:40:55,356 DEBUG            ec2:414  Getting tag 'filesystem' on resource 'vol-e80c1e37'
2016-03-01 22:40:55,356 DEBUG     filesystem:576  Adding Volume (id=None, size=0, snap=snap-7b992575) into Filesystem galaxyIndices FS
2016-03-01 22:40:55,356 DEBUG         volume:64   Volume service object created; creating the actual volume.
2016-03-01 22:40:55,521 DEBUG         volume:349  Creating a new volume of size '80' in zone 'us-east-1c' from snapshot 'snap-7b992575' for galaxyIndices FS.
2016-03-01 22:40:55,652 DEBUG         volume:360  Created a new volume of size '80' from snapshot 'snap-7b992575' with ID 'vol-8131235e' in zone 'us-east-1c' for galaxyIndices FS.
2016-03-01 22:40:55,652 DEBUG            ec2:389  Adding tag 'clusterName:cloudman-dev' to resource 'vol-8131235e'
2016-03-01 22:40:55,786 DEBUG            ec2:389  Adding tag 'bucketName:cm-3e863b9434af2d0e55c834cf200efc50' to resource 'vol-8131235e'
2016-03-01 22:40:55,903 DEBUG         volume:148  Getting snaps derived from volume vol-8131235e.
2016-03-01 22:40:57,878 DEBUG         master:2761 S&S: AS..Unstarted; ClouderaManager..Unstarted; Cloudgene..Unstarted; Galaxy..Unstarted; GalaxyReports..Unstarted; Migration..Completed; Nginx..OK; NodeJSProxy..Unstarted; PSS..Unstarted; Postgres..Unstarted; ProFTPd..Unstarted; Pulsar..Unstarted; Slurmctld..OK; Slurmd..OK; Supervisor..OK; transient_nfs FS..OK;
2016-03-01 22:40:57,878 DEBUG         master:2618 Monitor adding service 'PSS'
2016-03-01 22:40:57,878 DEBUG            pss:88   Galaxy service not running yet; will wait.
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/mnt/cm/cm/master.py", line 2790, in __monitor
    config_changed = self._start_services()
  File "/mnt/cm/cm/master.py", line 2621, in _start_services
    if service.add():
  File "/mnt/cm/cm/services/apps/pss.py", line 95, in add
    (break_srvc.get_full_name(), break_srvc.state, self.name))
UnboundLocalError: local variable 'break_srvc' referenced before assignment

2016-03-01 22:41:00,490 DEBUG         volume:159  Got snaps derived from volume vol-8131235e in 4.58710408211 seconds: []
2016-03-01 22:41:00,491 DEBUG         master:542  Adding a previously existing filesystem 'galaxyIndices' of kind 'snapshot'
2016-03-01 22:41:00,491 DEBUG         master:107  Adding a new file system service into the registry: galaxyIndices
2016-03-01 22:41:00,491 DEBUG       registry:111  Registering service galaxyIndices with the registry
2016-03-01 22:41:00,491 DEBUG         master:113  Activating service galaxyIndices
2016-03-01 22:41:00,492 DEBUG         master:157  ADD dependencies for service galaxyIndices
2016-03-01 22:41:00,492 DEBUG     filesystem:32   Instantiating Filesystem object galaxy with service roles: 'galaxyTools,galaxyData'
2016-03-01 22:41:00,492 DEBUG     filesystem:576  Adding Volume (id=vol-153123ca, size=0, snap=None) into Filesystem galaxy FS
2016-03-01 22:41:00,492 DEBUG         volume:169  Getting an update on volume vol-153123ca (<type 'unicode'>)
2016-03-01 22:41:00,492 DEBUG         volume:172  Retrieving a reference to the Volume object for ID vol-153123ca
2016-03-01 22:41:00,570 DEBUG         volume:191  Updating current `volume` object reference 'None' to a new one 'vol-153123ca'
2016-03-01 22:41:00,570 DEBUG         volume:199  For volume vol-153123ca (galaxy FS) set from_snapshot_id to snap-497a6b49
2016-03-01 22:41:00,575 DEBUG         volume:223  Volume vol-153123ca is not attached.
2016-03-01 22:41:00,575 DEBUG         volume:148  Getting snaps derived from volume vol-153123ca.
2016-03-01 22:41:02,775 DEBUG         volume:159  Got snaps derived from volume vol-153123ca in 2.19990801811 seconds: []
2016-03-01 22:41:02,776 DEBUG         master:542  Adding a previously existing filesystem 'galaxy' of kind 'volume'
2016-03-01 22:41:02,776 DEBUG         master:107  Adding a new file system service into the registry: galaxy
2016-03-01 22:41:02,776 DEBUG       registry:111  Registering service galaxy with the registry
2016-03-01 22:41:02,776 DEBUG         master:113  Activating service galaxy
2016-03-01 22:41:02,776 DEBUG         master:157  ADD dependencies for service galaxy
2016-03-01 22:41:02,776 DEBUG         master:559  Activating previously-available application services from an existing cluster config.
2016-03-01 22:41:02,776 DEBUG         master:113  Activating service Postgres
2016-03-01 22:41:02,776 DEBUG         master:157  ADD dependencies for service Postgres
2016-03-01 22:41:02,776 DEBUG         master:113  Activating service ProFTPd
2016-03-01 22:41:02,777 DEBUG         master:157  ADD dependencies for service ProFTPd
2016-03-01 22:41:02,777 DEBUG         master:113  Activating service GalaxyReports
2016-03-01 22:41:02,777 DEBUG         master:157  ADD dependencies for service GalaxyReports
2016-03-01 22:41:02,777 DEBUG         master:113  Activating service Slurmd
2016-03-01 22:41:02,777 DEBUG         master:157  ADD dependencies for service Slurmd
2016-03-01 22:41:02,777 DEBUG         master:113  Activating service Nginx
2016-03-01 22:41:02,777 DEBUG         master:157  ADD dependencies for service Nginx
2016-03-01 22:41:02,777 DEBUG         master:113  Activating service NodeJSProxy
2016-03-01 22:41:02,777 DEBUG         master:157  ADD dependencies for service NodeJSProxy
2016-03-01 22:41:02,778 DEBUG         master:113  Activating service Supervisor
2016-03-01 22:41:02,778 DEBUG         master:157  ADD dependencies for service Supervisor
2016-03-01 22:41:02,778 DEBUG         master:113  Activating service Slurmctld
2016-03-01 22:41:02,778 DEBUG         master:157  ADD dependencies for service Slurmctld
2016-03-01 22:41:02,778 DEBUG         master:113  Activating service Galaxy
2016-03-01 22:41:02,778 DEBUG         master:157  ADD dependencies for service Galaxy
2016-03-01 22:44:41,768 DEBUG            ec2:205  Gathering instance public keys (i.e., key pairs), attempt 0
2016-03-01 22:44:41,771 DEBUG            ec2:212  Got key pair: 'cloudman_key_pair
```
Also, _galaxyFS_ and _galaxyIndicesFS_ volumes are not attached to the master instance.

**Comment by dannon** [comment]
@hackdna This was fixed in 47096920 and will be in a forthcoming release.

**Comment by hackdna** [comment]
Thank you!
